### PR TITLE
upgrade gradle + config & remove unused dependencies

### PR DESCRIPTION
**Pull request by snopoke**
Review by commit.

I'm going to put this on staging for a while.

### CODE REVIEW

File: gradlew [hunk 1]
```diff
@@ -138,35 +156,30 @@ if $cygwin ; then
         else
             eval `echo args$i`="\"$arg\""
         fi
-        i=$((i+1))
+        i=`expr $i + 1`
```
Comment (calellowitz) [inline gradlew:159]: nit, i think `$()` is the preferred syntax for process subsitution

File: build.gradle [hunk 2]
```diff
@@ -41,64 +41,56 @@ idea {
...
-    compile group: 'org.springframework.boot', name: 'spring-boot'
-    compile group: 'org.springframework.boot', name: 'spring-boot-starter-web'
-    compile group: 'org.springframework.boot', name: 'spring-boot-starter-actuator'
-    compile group: 'org.springframework.boot', name: 'spring-boot-starter'
-    compile group: 'org.springframework.boot', name: 'spring-boot-starter-jdbc'
-    compile group: 'org.springframework.boot', name: 'spring-boot-autoconfigure'
-    compile group: 'org.springframework.boot', name: 'spring-boot-starter-data-jpa'
-    compile group: 'org.xerial', name: 'sqlite-jdbc'
-    compile group: 'redis.clients', name: 'jedis'
+    implementation group: 'com.fasterxml.jackson.core', name: 'jackson-databind'
```
Comment (calellowitz) [inline build.gradle:44]: Is this group name change required by the gradle upgrade?

**Review by calellowitz — COMMENTED** [review]
Two questions

Are these changes automatically generated? I noticed they make changes for platforms that I imagine you are not running. 

Is it an issue that we would have different gradle versions for the two repos, especially if they require different group names

File: build.gradle (hunk 2 above)
Comment (ctsims) [reply build.gradle:44]: Not quite, but soon. `compile` is headed for deprecation in version 7.

**Review by ctsims — APPROVED** [review]

**Comment by snopoke** [comment]
@calellowitz all the changes except those to `build.gradle` are made automatically via this command:
```
./gradlew wrapper --gradle-version 6.8
```

**Comment by snopoke** [comment]
I've tested this on staging:

* webapps & preview
* mobile user & web user & web user logging in as mobile user
* form completion & submission, sync, incomplete forms, app updates

**Comment by snopoke** [comment]
Jar size drops by 2.7MB (3%)

**Pull request merged**